components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -174,31 +177,36 @@ private void validateDateOfBirthClaimValue(String claimURI, String claimValue, U
 
         if (StringUtils.equalsIgnoreCase(DATE_OF_BIRTH_LOCAL_CLAIM, claimURI)) {
             String tenantDomain = IdentityTenantUtil.getTenantDomain(userStoreManager.getTenantId());
-            String dateOfBirthRegex = getDateOfBirthRegex(tenantDomain);
+            Map<String, String> dateOfBirthClaimProperties = getDateOfBirthClaimProperties(tenantDomain);
+            String dateOfBirthRegex = dateOfBirthClaimProperties.get(PROP_REG_EX);
             if (StringUtils.isEmpty(dateOfBirthRegex)) {
                 dateOfBirthRegex = DATE_OF_BIRTH_REGEX;
             }
             if (StringUtils.isNotEmpty(claimValue) && !claimValue.matches(dateOfBirthRegex)) {
-                throw new UserStoreException("Date of Birth doesn't match with the regex: " + dateOfBirthRegex);
+                String dateOfBirthRegexError = dateOfBirthClaimProperties.get(PROP_REG_EX_VALIDATION_ERROR);
+                if (StringUtils.isEmpty(dateOfBirthRegexError)) {
+                    dateOfBirthRegexError = DOB_REG_EX_VALIDATION_DEFAULT_ERROR;
+                }
+                throw new UserStoreClientException(dateOfBirthRegexError);
             }
         }
     }
 
-    private String getDateOfBirthRegex(String tenantDomain) {
+    private Map<String, String> getDateOfBirthClaimProperties(String tenantDomain) {
 
-        String dateOfBirthRegex = null;
+        Map<String, String> DateOfBirthClaimProperties = null;
         try {
             List<LocalClaim> localClaims =
                     SCIMCommonComponentHolder.getClaimManagementService().getLocalClaims(tenantDomain);
             for (LocalClaim localClaim : localClaims) {
                 if (StringUtils.equalsIgnoreCase(DATE_OF_BIRTH_LOCAL_CLAIM, localClaim.getClaimURI())) {
-                    dateOfBirthRegex = localClaim.getClaimProperties().get(PROP_REG_EX);
+                    DateOfBirthClaimProperties = localClaim.getClaimProperties();
                 }
             }
         } catch (ClaimMetadataException e) {
```

Comment:
log.error("Error while retrieving local claim meta data.", e);

Reply:
fixed in d9c5338